rotkehlchen/tests/unit/decoders/test_paraswap.py

```diff
@@ -0,0 +1,773 @@
+def test_simple_swap_token_fee(database, ethereum_inquirer, ethereum_accounts):
+    tx_hash = deserialize_evm_tx_hash(
+        '0xf4d28d1c46dde983fac73db8e651514fe7299edaf40870766462362b1b1a1d83',
+    )
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    user_address = ethereum_accounts[0]
+    timestamp = TimestampMS(1704797495000)
+    swap_amount = '138.2851'
+    received_amount = '1190.359719'
+    gas_fees = '0.0034595'
+    paraswap_fee = '11.308417'
+    expected_events = [EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=0,
+        timestamp=timestamp,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.SPEND,
+        event_subtype=HistoryEventSubType.FEE,
+        asset=A_ETH,
+        balance=Balance(amount=FVal(gas_fees)),
+        location_label=user_address,
+        notes=f'Burned {gas_fees} ETH for gas',
+        counterparty=CPT_GAS,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=1,
+        timestamp=timestamp,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.INFORMATIONAL,
+        event_subtype=HistoryEventSubType.APPROVE,
+        asset=A_ROUTE,
+        balance=Balance(amount=ZERO),
+        location_label=user_address,
+        notes=f'Revoke ROUTE spending approval of {user_address} by {PARASWAP_TOKEN_TRANSFER_PROXY}',  # noqa: E501
+        counterparty=None,
+        address=PARASWAP_TOKEN_TRANSFER_PROXY,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=2,
+        timestamp=timestamp,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.SPEND,
+        asset=A_ROUTE,
+        balance=Balance(amount=FVal(swap_amount)),
+        location_label=user_address,
+        notes=f'Swap {swap_amount} ROUTE in paraswap',
+        counterparty=CPT_PARASWAP,
+        address=PARASWAP_AUGUSTUS_ROUTER,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=3,
+        timestamp=timestamp,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.RECEIVE,
+        asset=A_USDC,
+        balance=Balance(amount=FVal(received_amount)),
+        location_label=user_address,
+        notes=f'Receive {received_amount} USDC as the result of a paraswap swap',
+        counterparty=CPT_PARASWAP,
+        address=PARASWAP_AUGUSTUS_ROUTER,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=116,
+        timestamp=timestamp,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.SPEND,
+        event_subtype=HistoryEventSubType.FEE,
+        asset=A_USDC,
+        balance=Balance(amount=FVal(paraswap_fee)),
+        location_label=user_address,
+        notes=f'Spend {paraswap_fee} USDC as paraswap fees',
```

Comment:
check the fee message in cowswap and use the same format as there. I don't know if you did but if not keep it consistent